### DISCUSSION TITLE
Struggling to understand baselines, versioning and custom registries

### DISCUSSION THREAD

**Discussion by senstar-nross** [microsoft/vcpkg#27535] [Q&A]
Can someone tell me if the following is true:

Every time I add a new package to a custom registry I need to:

1. Update the baseline in my vcpkg-configuration.json in order to use my new package, otherwise i get the error "The baseline does not contain the entry for port XXX"
2. If I want any existing dependencies to not upgrade to the versions currently specified in the baseline, I need to manually add overrides for all of them so they do not get bumped by the baseline
3. But if I just add a new version of an existing port I can just use it without moving baseline?

How does this make sense... If I had clairvoyance and filled my registry with all the packages that i need in the future I could just manage my versions normally.  But if I ever need to add a new package then I am forced to make sure to pin all my versions with overrides.

I am am quite confused by this.  No other package management system I have used has such a constraint...

Can someone please educate me???

**Comment by JackBoosY** [comment]
I think vcpkg should make more improvements about that.
